core/src/main/java/com/orientechnologies/orient/core/storage/ridbag/sbtree/OSBTreeCollectionManagerShared.java

```diff
@@ -186,4 +183,28 @@ public void clearPendingCollections() {
   public void clearChangedIds() {
     ODatabaseRecordThreadLocal.instance().get().getCollectionsChanges().clear();
   }
+
+  public boolean tryDelete(OBonsaiCollectionPointer collectionPointer, long delay) {
+    final CacheKey cacheKey = new CacheKey(storage, collectionPointer);
+    final Object lock = treesSubsetLock(cacheKey);
+    //noinspection SynchronizationOnLocalVariableOrMethodParameter
+    synchronized (lock) {
+      SBTreeBonsaiContainer container = treeCache.getQuietly(cacheKey);
+      assert container != null;
+      if (container.usagesCounter != 0 || container.lastAccessTime > System.currentTimeMillis() - delay) {
+        return false;
+      }
+
+      treeCache.remove(cacheKey);
+    }
+    OSBTreeBonsai<OIdentifiable, Integer> treeBonsai = this.loadSBTree(collectionPointer);
```

Comment:
I can not completely grasp it. At first, you remove tree rid bag from the cache. Then you load it into cache remove an instance, but keep a pointer into the cache again. Did I miss something?

Reply:
loadSBTree do not read from the cache, go direct to the disc, i remove it from the cache to delete it also from there

Reply:
Load sbtree https://github.com/orientechnologies/orientdb/pull/8968/commits/539125530e614cce80a2964f6fb6120b671bfbf6#L101 delegates to https://github.com/orientechnologies/orientdb/blob/539125530e614cce80a2964f6fb6120b671bfbf6/core/src/main/java/com/orientechnologies/orient/core/storage/ridbag/sbtree/OSBTreeCollectionManagerAbstract.java#L165 which populates cache back. May be I missed somethign ?

Reply:
yes you are right, i got it wrong, loadTree is acually the one that do not use the cache, loadSBTree does use it, so i'm going to make sure to use the one that do not access to the cache